lib/encoding.neon

```diff
@@ -0,0 +1,159 @@
+%|
+ |  File: encoding
+ |
+ |  Functions for encoding and decoding ASCII or Binary data in various formats.
+ |%
+IMPORT bitwise
+
+EXPORT base64Encode
+EXPORT base64Decode
+
+TYPE Mode IS ENUM
+    BYTE1
+    BYTE2
+    BYTE3
+    BYTE4
+END ENUM
+
+CONSTANT Base64EncodeTable: String := "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"
+
+CONSTANT Base64DecodeTable: Array<Number> := [
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,    % 0x00 - 0x0f
```

Comment:
There's no reason to store the value "255" in here, since it's numbers, not bytes. Use -1 for a flag saying the value is not valid (and that's much easier to check later).